rs/index/src/ivf/index.rs

```diff
@@ -36,6 +39,65 @@ pub struct Ivf<Q: Quantizer, DC: DistanceCalculator, D: IntSeqDecoder<Item = u64
     _decoder_marker: PhantomData<D>,
 }
 
+pub enum IvfType<Q: Quantizer> {
+    L2Plain(Ivf<Q, L2DistanceCalculator, PlainDecoder>),
+    L2EF(Ivf<Q, L2DistanceCalculator, EliasFanoDecoder>),
+}
+
```

Comment:
Do we really need this enum? Every time we just duplicate the call to a function down below

Reply:
we do, otherwise we need to plumb the template everywhere, very messy